stat/distuv/triangle.go

```diff
@@ -18,9 +18,9 @@ type Triangle struct {
 // NewTriangle constructs a new triangle distribution with lower limit a, upper limit b, and mode c.
 // Constraints are a < b and a ≤ c ≤ b.
 // This distribution is uncommon in nature, but may be useful for simulation.
-func NewTriangle(a, b, c float64) Triangle {
+func NewTriangle(a, b, c float64, src *rand.Rand) Triangle {
```

Comment:
The original API here was to do `t := NewTriangle(a, b, c); t.Source = src`. This was my fault; few of the types here have constructors and depend on direct setting of the source, but `Triangle` needs one because of the constraints between the parameters. I left the normal API approach of direct setting of the source. I'm not convinced that it needs to change, but I'm also not convinced that it shouldn't.